dagger-common/src/main/java/io/odpf/dagger/common/serde/parquet/SimpleGroupValidation.java

```diff
@@ -1,9 +1,94 @@
 package io.odpf.dagger.common.serde.parquet;
 
 import org.apache.parquet.example.data.simple.SimpleGroup;
+import org.apache.parquet.schema.GroupType;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
+import org.apache.parquet.schema.Type;
+
+import static org.apache.parquet.schema.Type.Repetition.OPTIONAL;
+import static org.apache.parquet.schema.Type.Repetition.REPEATED;
+import static org.apache.parquet.schema.Type.Repetition.REQUIRED;
 
 public class SimpleGroupValidation {
     public static boolean checkFieldExistsAndIsInitialized(SimpleGroup simpleGroup, String fieldName) {
         return simpleGroup.getType().containsField(fieldName) && simpleGroup.getFieldRepetitionCount(fieldName) != 0;
     }
+
+    /**
+     * This method checks if the map field inside the simple group is
+     * serialized using this legacy format:
+     * {@code
+     * <pre>
+     *     repeated group &lt;name&gt; {
+     *      &lt;repetition-type&gt; &lt;data-type&gt; key;
+     *      &lt;repetition-type&gt; &lt;data-type&gt; value;
+     *    }
+     * </pre>
+     * }
+     * The outer group is always repeated. key and value are constant field names.
+     *
+     * @param simpleGroup The SimpleGroup object inside which the map field is present
+     * @param fieldName   The name of the map field
+     * @return true, if the map structure follows the spec and false otherwise.
+     */
+    public static boolean checkIsLegacySimpleGroupMap(SimpleGroup simpleGroup, String fieldName) {
```

Comment:
There are mainly 2 reasons:
1. SRP: Even though it uses map, this function looks and behaves like a validator. It returns a boolean, yes or no answer.  I am giving this class a responsibility of validating simple groups against a field, that's it. Whether it's a map field, enum field, string field or what kind of validation doesn't quite matter. Maybe in the future if we start supporting simple group structs, we can also put validators here on whether it is a struct that is valid.
2. Modularity and Ease of Understanding: If you check this function carefully, it does a quite many `if` based checks and hence this needs to be tested for different corner cases. The unit tests are verbose. Here it is a public function and can be tested as it is, but when placed in MapHandler, this would ideally become a private function and hence the entire unit would need to be tested through the public interface of `transformFromParquet`. This method already has a lot of big tests right now and moving this validator function to MapHandler would result in a lot of hard to understand tests. The MapHandler class is already quite big and I don't want to make it bigger.

Reply:
As discussed, let's keep this for now